cmd/opampsupervisor/e2e_test.go

```diff
@@ -699,6 +716,31 @@ func TestSupervisorOpAMPConnectionSettings(t *testing.T) {
 	}, 10*time.Second, 500*time.Millisecond, "Collector did not connect to new OpAMP server")
 }
 
+func TestSupervisorCanStartWithoutServerThenConnectLater(t *testing.T) {
```

Comment:
Could we check that the Collector is started after the connection timeout but before the server is actually started in this test?

Reply:
added a test that we know the agent description. lmk if that looks fine.

Reply:
Thanks for adding this. I added a comment elsewhere, but I think we should check that we get a health message indicating the Collector is live.